helm/generated_examples/baremetal-affinity.yaml

```diff
@@ -172,3 +174,125 @@ spec:
         - name: local-storage
           hostPath:
             path: /mnt/disks
+---
+# Source: provisioner/templates/daemonset_windows.yaml
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: local-static-provisioner-win
+  namespace: default
+  labels:
+    helm.sh/chart: provisioner-2.5.0
+    app.kubernetes.io/name: provisioner
+    app.kubernetes.io/managed-by: Helm
+    app.kubernetes.io/instance: local-static-provisioner
+spec:
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: provisioner
+      app.kubernetes.io/instance: local-static-provisioner
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: provisioner
+        app.kubernetes.io/instance: local-static-provisioner
+      annotations:
+        checksum/config: 5a2fc47afffa2a982cff53af8b99da3694484be2a032fa8f6c723580d2d7fe19
+    spec:
+      serviceAccountName: local-static-provisioner
+      nodeSelector:
+        kubernetes.io/os: windows
+      tolerations:
+        # an empty key operator Exists matches all keys, values and effects
+        # which meants that this will tolerate everything
+        - operator: "Exists"
+      affinity:
+        
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: pool
+                operator: In
+                values:
+                - localssd
+      containers:
+        - name: provisioner
+          image: k8s.gcr.io/sig-storage/local-volume-provisioner:v2.4.0
+          env:
+          - name: MY_NODE_NAME
+            valueFrom:
+              fieldRef:
+                fieldPath: spec.nodeName
+          - name: MY_NAMESPACE
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.namespace
+          - name: JOB_CONTAINER_IMAGE
+            value: k8s.gcr.io/sig-storage/local-volume-provisioner:v2.4.0
+          ports:
+          - name: metrics
+            containerPort: 8080
+          volumeMounts:
+            - name: provisioner-config
+              mountPath: /etc/provisioner/config
+              readOnly: true
+            - name: provisioner-dev
+              mountPath: /dev
```

Comment:
Hi @mauriciopoppe have you verified that this local volume provisioner works on Windows? on Windows, there is no such `/dev`, the device discovery should be quite different on Windows IMO

Reply:
If I remember correctly `/dev` isn't read by the Windows impl because it doesn't have any mount points, during development I set the mount points in:

```
			- name: local-storage
              mountPath: /mnt/disks
              mountPropagation: HostToContainer
```

> the device discovery should be quite different on Windows IMO

This is true, I precreated mountPoints (symlinks to the actual device) in `/mnt/disks` in the host.

I run all the e2e tests successfully with CSI Proxy v1.1.0 locally and got the presubmits tests to pass in https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/pull/279 with Windows too, I think the remaining item was to bump kubernetes/kubernetes to use this version of CSI Proxy.

Reply:
@mauriciopoppe did you hit following error when start the provisioner:
```
I1007 14:38:40.991041    8780 common.go:348] StorageClass "local-disk" configured with MountDir "c:\\dev", HostDir "c:\\dev", VolumeMode "Filesystem", FsType "ext4", BlockCleanerCommand ["/scripts/shred.sh" "2"], NamePattern "nvme*"
I1007 14:38:41.544387    8780 main.go:66] Loaded configuration: {StorageClassConfig:map[local-disk:{HostDir:c:\dev MountDir:c:\dev BlockCleanerCommand:[/scripts/shred.sh 2] VolumeMode:Filesystem FsType:ext4 NamePattern:nvme*}] NodeLabelsForPV:[] UseAlphaAPI:false UseJobForCleaning:false MinResyncPeriod:{Duration:5m0s} UseNodeNameOnly:false LabelsForPV:map[] SetPVOwnerRef:false}
I1007 14:38:41.544387    8780 main.go:67] Ready to run...
I1007 14:38:41.544387    8780 common.go:425] Creating client using in-cluster config
I1007 14:38:41.593060    8780 main.go:132] Could not get node information (remaining retries: 2): Get "https://andy-aks1219-dns-a2701255.hcp.uksouth.azmk8s.io:443/api/v1/nodes/akswin000000": dial tcp: lookup andy-aks1219-dns-a2701255.hcp.uksouth.azmk8s.io: no such host
I1007 14:38:42.653329    8780 main.go:88] Starting controller
I1007 14:38:42.653329    8780 main.go:105] Starting metrics server at :8080
I1007 14:38:42.653329    8780 controller.go:47] Initializing volume cache
E1007 14:38:42.653329    8780 csi_proxy_windows.go:59] failed to connect to csi-proxy v1beta with error=open \\.\\pipe\\csi-proxy-volume-v1beta2: Access is denied.
F1007 14:38:42.653329    8780 controller.go:68] Error initializing VolumeUtil: open \\.\\pipe\\csi-proxy-volume-v1beta2: Access is denied.
```

Reply:
nvm, this PR should fix the issue: https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/pull/338